C/c4CAPI.cc

```diff
@@ -1504,18 +1504,21 @@ void c4queryenum_release(C4QueryEnumerator *e) noexcept {
 
 
 C4QueryObserver* c4queryobs_create(C4Query *query, C4QueryObserverCallback cb, void *ctx) noexcept {
-    auto fn = [cb,ctx](C4QueryObserver *obs) {
-        cb(obs, obs->query(), ctx);
-    };
-    return new C4QueryObserverImpl(query, fn);
+    C4Error error;
+    return tryCatch<C4QueryObserver*>(&error, [&] {
+        auto fn = [cb,ctx](C4QueryObserver *obs) {
+            cb(obs, obs->query(), ctx);
+        };
+        return C4QueryObserverImpl::newQueryObserver(query, fn).detach();
+    });
 }
 
 void c4queryobs_setEnabled(C4QueryObserver *obs, bool enabled) noexcept {
     obs->setEnabled(enabled);
 }
 
 void c4queryobs_free(C4QueryObserver* obs) noexcept {
-    delete obs;
+    c4base_release(obs);
```

Comment:
We'd want to alias c4queryobs_release for public use in the future.  For a VF it is ok.

Reply:
Yes. For released version,  such as 3.1 and 3.2, we keep the name. For Boron/master branch, when I create a port ticket, I will mention it to use stand naming for RefCounted, _new and _release.